### DISCUSSION TITLE
Quadratic equality constraints?

### DISCUSSION THREAD

**Discussion by RylanSchaeffer** [qpsolvers/qpsolvers#241] [Q&A] [answered]
Apologies if I missed this in the documentation, but how does one specify a quadratic equality constraint? i.e. something like:

$$x^T Q x = 1$$

where $Q$ is a positive semidefinite matrix.

**Comment by stephane-caron** [comment, marked as answer]
A quadratic equality constraint would give you a [Quadratically constrained quadratic program](https://en.wikipedia.org/wiki/Quadratically_constrained_quadratic_program) (QCQP), which is not the same class of problems as [QP](https://en.wikipedia.org/wiki/Quadratic_programming). All equality constraints in a QP are linear.

For QCQPs, you can check out: 

- The [HPIPM](https://github.com/giaf/hpipm) solver, which can solve QCQPs.
- This discussion: [I have a non-convex quadratic program, is there a solver I can use?](https://github.com/qpsolvers/qpsolvers/discussions/240)
